flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/TaskExecutorRegistration.java

```diff
@@ -55,6 +56,10 @@ public class TaskExecutorRegistration implements Serializable {
     /** The task executor total resource profile. */
     private final ResourceProfile totalResourceProfile;
 
+    /** ID of the node where the TaskManager is located. */
+    private final String nodeId;
```

Comment:
This change seems to serve another purpose which is different from adding `nodeId` to `TaskManagerLocation`.

Reply:
I have extracted it as a separate commit。